packages/preview-service/services/screenshot.ts

```diff
@@ -0,0 +1,120 @@
+import { reduce } from 'lodash'
+import {
+  shouldBeHeadless,
+  getPuppeteerUserDataDir,
+  getChromiumExecutablePath
+} from '../utils/env'
+import type { Logger } from 'pino'
+import { PuppeteerClientInterface } from '../clients/puppeteer'
+import type { ObjectIdentifier } from '../domain/domain'
+
+export type GetScreenshot = (
+  params: ObjectIdentifier
+) => Promise<{ [key: string]: string } | null>
+
+export const getScreenshotFactory =
+  (deps: {
+    puppeteerClient: PuppeteerClientInterface
+    logger: Logger
+    serviceOrigin: string
+  }): GetScreenshot =>
+  async (params) => {
+    const objectUrl = `${deps.serviceOrigin}/streams/${params.streamId}/objects/${params.objectId}`
+    const launchParams = {
+      //TODO as the launch params are touching environment variables, should they be dependency injected?
+      headless: shouldBeHeadless(),
```

Comment:
I noticed that all of these 3 env vars have defaults, altho we probably don't want to pass in anything at least in dev environments, so it does its own algo to figure out where to find chromium, where to put stuff etc.